### ISSUE TITLE
saving

### ISSUE THREAD

**Issue by cirelesna** [cirelesna/pe#8]
![image.png](https://raw.githubusercontent.com/cirelesna/pe/main/files/e3ddf3c2-0a66-4854-996f-bdff3e7a84ab.png)

I ctrl C after adding a run which resulted in my data integrity being compromised instead it should be saved after each command

**Comment by nus-se-script** [comment]
# Team's Response
User has **failed to read** the UG **_carefully_**. It was clearly stated tampering was **not allowed**. Furthermore, the user **should have kept a backup file** if they were **truly concerned** about data preservation. Finally, it is mentioned that once the app has exited **properly with** `exit` only then would the save and update be done to the .txt file:

![image.png](https://raw.githubusercontent.com/nus-cs2113-AY2324S2/pe/master/files/6e59e283-2ead-4ea0-946d-771c923fd37f.png)


Questions related to data storage have been addressed in FAQ:

![image.png](https://raw.githubusercontent.com/nus-cs2113-AY2324S2/pe/master/files/bdbdfb7c-e2c1-48c1-bdd7-a3dd09645233.png)

![image.png](https://raw.githubusercontent.com/nus-cs2113-AY2324S2/pe/master/files/801d52f4-7dc2-43f2-ac75-03d936dd76c3.png)

![image.png](https://raw.githubusercontent.com/nus-cs2113-AY2324S2/pe/master/files/e9ed38b7-a0ce-4170-962c-daad0367bd0d.png)

Finally you have also been warned of the delete : " For safety and security reasons, PulsePilot will automatically delete any
remaining data files before exiting the application."

In conclusion, this is an **insincere/random evaluation** and as stated [here in the CS2113 website](https://nus-cs2113-ay2324s2.github.io/website/admin/tp-pe.html) under **PE Overview**, such bug reports are unacceptable. 

![image.png](https://raw.githubusercontent.com/nus-cs2113-AY2324S2/pe/master/files/9cd0a7ca-cfeb-4480-8c80-84437fd09e2c.png)

We will flag this bug "report" to the tutors for further escalation. Thanks for your time Ansel :)

## The 'Original' Bug

[The team marked this bug as a duplicate of the following bug]

># Complete wipe of data
>
>![image.png](https://raw.githubusercontent.com/nus-cs2113-AY2324S2/pe/master/files/dc12a23d-4424-4c38-9666-7d10fae1b321.png)
>
>
>![image.png](https://raw.githubusercontent.com/nus-cs2113-AY2324S2/pe/master/files/64ecce2c-75d3-4ff4-9895-960a56deb31e.png)
>
>
>While I intentially corrupt the data and re-enter the app, I was made quit forcibly. All my data were lost, I think the app could give the user a second option to vet through the text file themselves as the data were somewhat valuable to them.
>
>
>
>
>
>
><hr>
><sub>[original: nus-cs2113-AY2324S2/pe-interim#1343] [original labels: type.FeatureFlaw severity.High]</sub>
>
>
>

## Their Response to the 'Original' Bug

[This is the team's response to the above 'original' bug]

>User has **failed to read** the UG **_carefully_**. It was clearly stated tampering was **not allowed**. Furthermore, the user **should have kept a backup file** if they were **truly concerned** about data preservation. Finally, looking through the text file is allowed and **could have been done without tampering any of the data** at any point after the app has exited **properly**:
>
>![image.png](https://raw.githubusercontent.com/nus-cs2113-AY2324S2/pe/master/files/f192d753-4386-4c71-b908-4f96774be7e1.png)
>
>Questions related to data storage have been addressed in FAQ:
>
>![image.png](https://raw.githubusercontent.com/nus-cs2113-AY2324S2/pe/master/files/bdbdfb7c-e2c1-48c1-bdd7-a3dd09645233.png)
>
>![image.png](https://raw.githubusercontent.com/nus-cs2113-AY2324S2/pe/master/files/801d52f4-7dc2-43f2-ac75-03d936dd76c3.png)
>
>![image.png](https://raw.githubusercontent.com/nus-cs2113-AY2324S2/pe/master/files/e9ed38b7-a0ce-4170-962c-daad0367bd0d.png)
>
>Finally you have also been warned of the delete : " For safety and security reasons, PulsePilot will automatically delete any
>remaining data files before exiting the application."
>
>In conclusion, this is an **insincere/random evaluation** and as stated [here in the CS2113 website](https://nus-cs2113-ay2324s2.github.io/website/admin/tp-pe.html) under **PE Overview**, such bug reports are unacceptable. 
>
>![image.png](https://raw.githubusercontent.com/nus-cs2113-AY2324S2/pe/master/files/9cd0a7ca-cfeb-4480-8c80-84437fd09e2c.png)
>
>We will flag this bug "report" to the tutors for further escalation. Thanks for your time :)
# Items for the Tester to Verify
## :question: Issue duplicate status

Team chose to mark this issue as a duplicate of another issue (as explained in the _**Team's response**_ above)

- [x] I disagree

**Reason for disagreement:** The issue the team has said is a duplicate of mine involves intentionally corrupting the data txt file, however the issue I highlighted occurs when the user exits the application using ctrlC or closing the CLI window

<catcher-end-of-segment><hr>
## :question: Issue response

Team chose [`response.Rejected`]

- [x] I disagree

**Reason for disagreement:** The reason given for rejection is that users should only exit with the exit command so as that data is saved and hash is generated. It is also stated that if the user wishes to preserve their data they should keep a backup elsewhere. However, I feel that the onus should not be placed on the user and cause their data to be lost just from the application crashing especially since this is done on purpose.

For example if after exiting the first session, the user starts up a second session inputting new information if they were to close the application without the exit command they would lose all their data. Even if the user kept a copy of the data they would only have data from the first session. The data from the second session would be completely lost. 

I had read the UG guide and disagree that the data files are only saved only after exit command and rather it should be saved and hash updated after each command. Thereby, even if the application closes data will be saved. Furthermore users will still be unable to tamper the data files.

Relying on the users to always execute the "exit" command introduces a risk of data loss that could be mitigated with auto-save after each command.

<catcher-end-of-segment><hr>
## :question: Issue severity

Team chose [`severity.VeryLow`]
Originally [`severity.High`]

- [x] I disagree

**Reason for disagreement:** The risk of data loss is a major inconvenience to users affecting usage and is not a purely cosmetic issue. 

<catcher-end-of-segment><hr>
